docs/guide/file_management/file_management.md

```diff
@@ -0,0 +1,147 @@
+    $fileContent = $this->ioService->getFileContents($file);
+    
+    // ...
+}
+```
+
+## Handling binary files
+
+[[= product_name =]] supports multiple binary file handling mechanisms by means of an `IOHandler` interface. This feature is used by the [BinaryFile](../../api/field_types_reference/binaryfilefield.md), [Media](../../api/field_types_reference/mediafield.md) and [Image](../../api/field_types_reference/imagefield.md) Field Types.
+
+### Native IO handler
+
+The IO API is organized around two types of handlers, both used by the IOService:
+
+- `Ibexa\Core\IO\IOMetadataHandler`: stores and reads metadata (such as, validity, size)
+- `Ibexa\Core\IO\IOBinarydataHandler`: stores and reads the actual binary data
+
+You can configure IO handlers using semantic configuration. IO handlers are configurable per SiteAccess.
+See the default configuration:
+
+``` yaml
+ibexa:
+    system:
+        default:
+            io:
+                metadata_handler: dfs
+                binarydata_handler: nfs
+```
+The adapter is the *driver* used by Flysystem to read/write files. Adapters are declared using `oneup_flysystem`. 
+Metadata and binary data handlers are configured under `ibexa_io`. See below the configuration for the default handlers. It declares a metadata handler and a binary data handler, both labeled `default`. Both handlers are of type `flysystem`, and use the same Flysystem adapter, labeled `default` as well.
+
+``` yaml
+ibexa_io:
+    binarydata_handlers:
+        nfs:
+            flysystem:
+                adapter: nfs_adapter
+    metadata_handlers:
+        dfs:
+            legacy_dfs_cluster:
+                connection: doctrine.dbal.dfs_connection
+```
+
+The nfs_adapter's directory is based on your site settings, and will automatically be set to `$var_dir$/$storage_dir$` (for example, `/path/to/ibexa/public/var/site/storage`).
+
+#### Permissions of generated files
+
+``` yaml
+ibexa:
+    system:
+        default:
+            io:
+                permissions:
+                    files: 0750 #default is 0644
+                    directories: 0640 #default is 0755
+```
+
+Both `files` and `directories` are optional.
+
+Default values:
+
+- 0644 for files
+- 0755 for directories
+
+!!! note
+
+    Make sure to configure permissions using a number and **not** a string. If you write "0644" it will **not** be interpreted by PHP as an octal number, and unexpected permissions is applied.
+
+!!! note
+
+    As SiteAccess configuration Flysystem's native Local NFS adapter is not supported, the following
+    configuration should be used:
+
+    ``` yaml
+    oneup_flysystem:
+        adapters:
+            nfs_adapter:
+                custom:
+                    service: ibexa.io.nfs.adapter.site_access_aware
+    ```
+
+
+### Native Flysystem handler
+
+[league/flysystem](http://flysystem.thephpleague.com/) (along with [FlysystemBundle](https://github.com/1up-lab/OneupFlysystemBundle/)) is an abstract file handling library.
```

Comment:
How about switching this and the next sentence? So we first know that Ibexa uses this package and them we explain what it is.